src/main/kotlin/com/jahnelgroup/controller/settings/user/preferences/UpdatePreferencesController.kt

```diff
@@ -0,0 +1,49 @@
+package com.jahnelgroup.controller.settings.user.profile
+
+import com.jahnelgroup.domain.context.UserContextService
+import com.jahnelgroup.domain.user.UserService
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.validation.BindingResult
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import javax.validation.Valid
+
+@Controller
+class UpdatePreferencesController(
+        private var userService: UserService,
+        private var userContextService: UserContextService){
+
+    // Forward is the same as redirect but the URL remains the same
+    @GetMapping("/settings/preferences")
+    fun profile() = "forward:/settings/${userContextService.currentUsername()}/preferences"
+
+    @PreAuthorize("hasRole('ROLE_ADMIN') || #user == authentication.name")
+    @GetMapping("/settings/{user}/preferences")
+    fun profile(model: Model, @PathVariable user: String): String{
+        val u = userService.findByUsername(user)
+        model.addAttribute("user", u)
+        model.addAttribute("updatePreferencesForm",
+                UpdatePreferencesForm(preferences = u.preferences))
+        return "layouts/settings/user/preferences"
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN') || #user == authentication.name")
+    @PostMapping("/settings/{user}/preferences")
+    fun updatePreferences(model: Model, @PathVariable user: String,
```

Comment:
I like that you created a new Controller and methods to handle these actions but the main method here updatePreferences ins't really doing anything. It'd be good if you added something to set a preference for like.... "Sort task alphabetically"